File: iOS/ssabree/ssabree/Core/Network/BoardService.swift
import Foundation

// MARK: - Board Service Protocol

protocol BoardService {
    func getBoards() async throws -> [BoardResponse]
    func getNotice() async throws -> BoardNoticeResponse
}

// MARK: - Board Service Implementation

final class BoardServiceImpl: BoardService {
    private let apiClient: APIClient

    init(apiClient: APIClient = .shared) {
        self.apiClient = apiClient
    }

    func getBoards() async throws -> [BoardResponse] {
        let endpoint = APIEndpoint(path: "/api/boards", method: .GET, requiresAuth: true)
        return try await apiClient.request(endpoint: endpoint, body: nil as String?, queryItems: nil)
    }

    func getNotice() async throws -> BoardNoticeResponse {
        let endpoint = APIEndpoint(path: "/api/boards/notice", method: .GET, requiresAuth: true)
        return try await apiClient.request(endpoint: endpoint, body: nil as String?, queryItems: nil)
    }
}

File: android/ssabree/app/src/main/java/com/ssafy/ssabree/core/datasource/remote/BoardService.kt
package com.ssafy.ssabree.core.datasource.remote

import com.ssafy.ssabree.core.datasource.remote.model.BoardResponse
import com.ssafy.ssabree.core.datasource.remote.model.BoardNoticeResponse
import retrofit2.http.GET

interface BoardService {
    @GET("/api/boards")
    suspend fun getBoards(): List<BoardResponse>

    @GET("/api/boards/notice")
    suspend fun getNotice(): BoardNoticeResponse
}
